plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -852,6 +854,35 @@ entryExitTracing lamName displayName e ty =
    an unfolding.
 -}
 
+takeHaskellList
```

Comment:
This is a strange name. Perhaps "compileCaseIntegerBranches"?

Reply:
Naming was intentionally made a bit generic because this can be used for other things as well; For example, 

`ands :: [Bool] -> Bool`

that compiles given entries into `p1 && p2 && p3 ... && pn`. 

I'll see if I can come up with a better name